### ISSUE TITLE
supporting local path in --extra-data

### ISSUE THREAD

**Issue by kujeger** [flatpak/flatpak#713]
I'm using flatpak to run various games using the linux installers from GOG.
These installers can sometimes be quite large, and seem to be a good use case for the extra-data option.

However, extra-data currently only supports http(s) addresses, and these installers are not freely available for download. Being able to refer to local paths would make it possible to use flatpak to run these games without having to keep a full copy of the game in the (possibly local) repository.

**Comment by alexlarsson** [comment]
I don't see how you could use these outside your own machine though. That doesn't seem very useful.

**Comment by TingPing** [comment]
This is probably overkill for a limited use-case. But flatpak could expose a way for users to select data.

**Comment by Barthalion** [comment]
Any plans to revisit this now, when i386 multiarch extension has been added? I would love to get rid of these packages shipped by my distro, but I need it for games from Humble Bundle/GOG stores.

**Comment by manuq** [comment]
The extra-data files are downloaded to BASE_DIR/extra-data/SHASUM/

And flatpak will skip the download if the file is already there.

So you can just put the file there inside a directory with the corresponding checksum. In my case BASE_DIR is  ~/.local/share/flatpak/

**Comment by gasinvein** [comment]
I use `python3 -m http.server` to ease testing of `apply-extra`, when the `extra-data` is too large or original source is too slow.

**Comment by bam80** [comment]
Unfortunately, right now absence of this feature prevents my flatpak to be accepted: https://github.com/flathub/flathub/pull/994#issuecomment-492452901.
This is very sad for me because I put much efforts to this.
So I'm asking devs, please, consider to implement this feature.

**Comment by bam80** [comment]
Seems the restriction with `http` uris only in --extra-data was done intentionally due security reasons: https://github.com/flatpak/flatpak/blob/2b939282859143b0a86beed90574a791e877d4e1/common/flatpak-dir.c#L4798

I wonder if we could softening it by allowing file URIs with only certain dirs here, e.g XDG_DOWNLOAD_DIR?

**Comment by Mikenux** [comment]
@bam80: I agree with @TingPing on user experience. Rather, it is about adding a specific `filesystem` permission to an app, at least if it cannot work with the document portal. This seems to be more of a portal thing. There is already a discussion about this: https://github.com/flatpak/xdg-desktop-portal/discussions/1405.
